plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,644 @@
+package ravendb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// defaultURL will set a default value that corresponds to the default value
+// used by RavenDB
+const defaultURL = "http://localhost:8080"
+
+// Default http timeouts
+const defaultResponseHeaderTimeout = 3
+const defaultClientTimeout = 4
+
+type gatherFunc func(r *RavenDB, acc telegraf.Accumulator)
+
+var gatherFunctions = []gatherFunc{gatherServer, gatherDatabases, gatherIndexes, gatherCollections}
+
+// RavenDB defines the configuration necessary for gathering metrics,
+// see the sample config for further details
+type RavenDB struct {
+	URL  string `toml:"url"`
+	Name string `toml:"name"`
+
+	ResponseHeaderTimeout internal.Duration `toml:"header_timeout"`
+	ClientTimeout         internal.Duration `toml:"client_timeout"`
```

Comment:
Do we really need this fine-grained timeout definition or can we just work with the `client_timeout`? If so, please rename the option to `timeout`.

Reply:
done - initially I took that from rabbitmq and thought it is standard way of doing that.

Reply:
Yeah no problem. Please mark as resolved.